### PR TITLE
new: gha release ci

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
Prow is failing :/

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
```
Comment (leogr) [inline .github/workflows/release.yml:6]: The original regex includes a dash (`-`) which is not reported here.

I can't remember why we needed that. Just noticed the difference.
cc @jasondellaluce

**Review by leogr — COMMENTED** [review]
Just a doubt. Otherwise, SGTM!

File: .github/workflows/release.yml (hunk 1 above)
Comment (jasondellaluce) [reply .github/workflows/release.yml:6]: We likely wanted to support the dash to have tags for release candidates such as `v0.7.0-rc1`. Not sure if we want those tags to be released with GoReleaser though!

Comment (FedeDP) [reply .github/workflows/release.yml:6]: Well i think we can easily add support for that in the regex, right? Let's do it now and forget about this :)

**Review by jasondellaluce — APPROVED** [review]
/approve

File: .github/workflows/release.yml (hunk 1 above)
Comment (FedeDP) [reply .github/workflows/release.yml:6]: Indeed i think we can post-pone this since we are not going to use it rn!

**Pull request merged**